lib/jackalope/work_list.ex

```diff
@@ -1,128 +1,73 @@
 defmodule Jackalope.WorkList do
-  @moduledoc false
-
-  require Logger
-  @default_max_size 100
-
-  defstruct items: [],
-            pending: %{},
-            max_size: @default_max_size
-
-  @type t() :: %__MODULE__{items: list(), max_size: non_neg_integer()}
-
-  @spec new(list(), non_neg_integer()) :: t()
-  def new(items, max_size \\ @default_max_size) when max_size > 0 do
-    %__MODULE__{items: items, max_size: max_size}
-  end
-
-  @spec push(t(), any) :: t()
-  def push(work_list, item) do
-    updated_items =
-      [item | work_list.items]
-      |> bound(work_list.max_size)
-
-    %__MODULE__{work_list | items: updated_items}
-  end
-
-  @spec prepend(t(), list) :: t()
-  def prepend(work_list, items) when is_list(items) do
-    updated_items =
-      (items ++ work_list.items)
-      |> bound(work_list.max_size)
-
-    %__MODULE__{work_list | items: updated_items}
+  @moduledoc """
+  Behaviour for work lists and dispatcher to implementations.
+  """
+
+  @type t() :: any()
```

Comment:
This is the work list state. I didn't think it was a type. 

```suggestion
  @type state() :: any()
```

Could you add docs? I think that will help me understand intent.

Reply:
The intent is `instance()`, the actual work list instance.